cmd/system/aws_cli.go

```diff
@@ -0,0 +1,317 @@
+	command.Flags().StringP("work-dir", "w", "", "Working directory that installer files should be copied to (current directory if not supplied)")
+	command.Flags().Bool("run-installer", false, "Whether or not arkade should run the downloaded installer")
+
+	command.PreRunE = func(cmd *cobra.Command, args []string) error {
+		_, err := cmd.Flags().GetString("path")
+		if err != nil {
+			return err
+		}
+
+		_, err = cmd.Flags().GetBool("progress")
+		if err != nil {
+			return err
+		}
+
+		_, err = cmd.Flags().GetString("work-dir")
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		installPath, _ := cmd.Flags().GetString("path")
+		version, _ := cmd.Flags().GetString("version")
+		progress, _ := cmd.Flags().GetBool("progress")
+		workDir, _ := cmd.Flags().GetString("work-dir")
+		runInstaller, _ := cmd.Flags().GetBool("run-installer")
+		fmt.Printf("Installing AWS CLI to %s\n", installPath)
+
+		installPath = strings.ReplaceAll(installPath, "$HOME", os.Getenv("HOME"))
+
+		if err := os.MkdirAll(installPath, 0755); err != nil && !os.IsExist(err) {
+			fmt.Printf("Error creating directory %s, error: %s\n", installPath, err.Error())
+		}
+
+		arch, osVer := env.GetClientArch()
+
+		if cmd.Flags().Changed("arch") {
+			arch, _ = cmd.Flags().GetString("arch")
+		}
+
+		if version == "" {
+			v, err := getAWSCLIVersion("aws", "aws-cli")
+			if err != nil {
+				return err
+			}
+			version = v
+		}
+
+		fmt.Printf("Installing version: %s for: %s\n", version, arch)
+
+		awsCliTool := get.Tool{
+			Owner:   "amazonaws",
+			Repo:    "awscli",
+			Name:    "awscli",
+			Version: version,
+			URLTemplate: `
+			{{$version := .Version}}
+			{{$base := printf "https://%s.%s.com" .Repo .Owner}}
+
+			{{$ext := "zip"}}
+			{{$uri := printf "%s-exe-linux-%s" .Repo .Arch}}
+
+			{{ if HasPrefix .OS "Ming" }}
+			{{$uri = "AWSCLIV2"}}
+			{{$ext = "msi"}}
+			{{ else if eq .OS "Darwin" -}}
+			{{$uri = "AWSCLIV2"}}
+			{{$ext = "pkg"}}
+			{{ end -}}
+
+			{{$base}}/{{$uri}}-{{$version}}.{{$ext}}
+		`,
+		}
+
+		dUrl, err := awsCliTool.GetURL(osVer, arch, version, !progress)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloading from: %s\n", dUrl)
+
+		outPath, err := get.DownloadFileP(dUrl, progress)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloaded to: %s\n", outPath)
+
+		f, err := os.OpenFile(outPath, os.O_RDONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		if workDir == "" {
+			cwd, err := os.Getwd()
+			if err != nil {
+				return err
+			}
+			workDir = cwd
+		}
+
+		fmt.Printf("Copying file to: %s\n", workDir)
+
+		filename := filepath.Base(outPath)
+		if _, err = get.CopyFileP(
+			outPath,
+			fmt.Sprintf("%s/%s", workDir, filename), readWriteExecuteEveryone,
+		); err != nil {
+			return err
+		}
+
+		extension := filepath.Ext(outPath)
+		if extension == ".zip" {
```

Comment:
The `get` package already have a method for this on the `Tool` struct: `IsArchive` I would rather use that.